src/components/KInput/KInput.vue

```diff
@@ -384,6 +386,12 @@ $kInputSlotSpacing: var(--kui-space-40, $kui-space-40); // $kSelectInputSlotSpac
         width: $kInputIconSize !important;
       }
 
+      // enhance the experience for most common cases that icon only slots should not
+      // prevent the input from being focused by click on the icon
+      &:has(> #{$kongponentsKongIconSelector}:not(button):not([role="button"]):only-child) {
+        pointer-events: none;
+      }
```

Comment:
I checked our usage of the `before`/`after` slot across codebases, in all cases for `before` slot we are using single icon and in `after` slot there's only [one place](https://github.com/Kong/public-ui-components/blob/83a06bbe5e57241d3fb4d93f95ec834c9f150572/packages/entities/entities-shared/src/components/entity-filter/EntityFilter.vue#L14-L20) we are using clickable icon that does not have a `role="button"` set. We should fix that first.